posts/sgx.md

```diff
@@ -0,0 +1,164 @@
+# SGX
+## Introduction
+
+Intel SGX is a set of instructions used to enhance the security of application code and data, which gives the users a great level of protection. Intel SGX helps to keep the users’ sensitive data from being accesed by creating a trusted execution environment within memory.
+
+Sensitive data may be medical records, financial assets, passwords, encryption keys, biometric identification. In summary, any information that could cause harm if it's revealed.
+
+In order to abstract from operating system security vulnerabilities that host devices may have, essential parts of the project are deployed in a specific trusted execution environment (TEEs).
+
+TEEs can be thought of as processes that run "isolated" from the operating system and higher layers in a safe part of the CPU. TEEs are intended to ensure the integrity of data, code and data privacy.
+
+Intel SGX is an Intel Instruction Set Architecture (ISA) extension with TEEs support. The environments are called enclaves.
+
+Some important items:
+
+* **It's not possible to read nor write the enclave's memory space from outside the enclave**, regardless of the privilege level and CPU mode.
+* In production, it's not possible to debug enclaves by software nor hardware.
+* Entering the enclave via function calls, jumps or stack/register manipulation is not possible. To do so you have to use a specific CPU instruction which also does some safety checks.
+* **Enclave's memory is encrypted**, and the key used changes on every power cycle. It's stored within the CPU and is not accessible.
+
+![](https://i.imgur.com/eUXGLjZ.png)
+
+
+## Remote Attestation and Encryption research summary
```

Comment:
This should have a title with more punch like: "SGX's security features: remote attestation and encryption"